src/services/users/types.ts

```diff
@@ -916,33 +946,79 @@ export interface UsersService {
     requestBody: Hash,
     options?: OptionsBase
   ): Promise<boolean>;
+
   /**
-   * Request a password reset
+   * Request a password reset.
+   *
+   * An email is send to the targeted user with a token and instructions to reset their password.
+   * The token should be used to complete the password reset via `exh.users.validatePasswordReset`.
+   *
+   * The email send to the user is the email template configured by `passwordResetEmailTemplateId`.
+   * The template receives a 40 hexadecimal character hash in the `content.reset_hash` variable.
    *
    * Permission | Scope | Effect
    * - | - | -
    * none |  | Everyone can use this endpoint
-   * @param email
-   * @returns {boolean} true on success
+   *
    * @throws {EmailUnknownError}
    * @throws {NotActivatedError}
+   * @throws {IllegalStateError} Attempting to use either `passwordResetEmailTemplateId` or `passwordResetPinEmailTemplateId` while not configured. See `exh.users.setEmailTemplates`.
+   * @throws {ForgotPasswordRequestLimitError} The maximum allowed consecutive forgot password requests is reached
+   * @throws {ForgotPasswordRequestTimeoutError} Forgot password request too short after the previous one
    */
   requestPasswordReset(email: string, options?: OptionsBase): Promise<boolean>;
+
+  /**
+   * Request a password reset.
+   *
+   * An email is send to the targeted user with a token and instructions to reset their password.
+   * The token should be used to complete the password reset via `exh.users.validatePasswordReset`.
+   *
+   * By default the email send to the user is the email template configured by `passwordResetEmailTemplateId`.
+   * The template receives a 40 hexadecimal character hash in the `content.reset_hash` variable.
+   *
+   * If enabled, a pin code can be used rather than a hash.
+   * The pin code mode must be enabled by the `enablePinCodeForgotPasswordRequests` verification setting.
+   * To use the pin code mode, the `mode` field can be set to `pin_code`.
+   * Then the email send to the user is the email template configured by `passwordResetPinEmailTemplateId`.
+   * The pin code template receives a 8 digit pin code in the `content.pin_code` variable.
```

Comment:
I know these texts are not ideal. The goal here was just to provide the facts. A nice to read text should be coming up later 😬 (Spelling mistakes and logical corrections are more then welcome though)

Reply:
It's more than good enough 😂